BlueprintUICommonControls/Sources/Box.swift

```diff
@@ -214,34 +204,160 @@ extension Box.ShadowStyle {
             return color
         }
     }
-    
-    
 }
 
-fileprivate final class BoxView: UIView {
+
+extension UIRectCorner {
     
-    let contentView = UIView()
+    /// `CACornerMask` is based on the macOS coordinate system, which starts in the top left, not the bottom left like iOS.
+    /// You know when you try to plug in a USB connector and it takes 3 tries and you end up exactly where you started? Anyways.
     
-    override init(frame: CGRect) {
-        super.init(frame: frame)
-        contentView.frame = bounds
-        addSubview(contentView)
+    var toCACornerMask : CACornerMask {
+        var mask = CACornerMask()
+        
+        if self.contains(.topLeft) {
+            mask.formUnion(.layerMinXMinYCorner)
+        }
+        
+        if self.contains(.topRight) {
+            mask.formUnion(.layerMaxXMinYCorner)
+        }
+        
+        if self.contains(.bottomRight) {
+            mask.formUnion(.layerMaxXMaxYCorner)
+        }
+        
+        if self.contains(.bottomLeft) {
+            mask.formUnion(.layerMinXMaxYCorner)
+        }
+        
+        return mask
     }
+}
+
+
+extension Box {
     
-    required init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+    fileprivate final class View : UIView {
+        
+        let contentView = UIView()
+        
+        var model : ViewModel {
+            didSet {
+                guard oldValue != self.model else { return }
+             
+                self.needsUpdateFromModel = true
+                self.setNeedsLayout()
+            }
+        }
+        
+        init(frame: CGRect, model : ViewModel) {
+            
+            self.model = model
+            
+            super.init(frame: frame)
+            
+            contentView.frame = bounds
+            addSubview(contentView)
+        }
+        
+        required init?(coder aDecoder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+        
+        
+        private var needsUpdateFromModel : Bool = true
+        private var lastSize : CGSize? = nil
+        
+        override func layoutSubviews() {
+            super.layoutSubviews()
+            
+            contentView.frame = bounds
+
+            if self.lastSize != self.bounds.size || self.needsUpdateFromModel {
+                self.lastSize = self.bounds.size
+                self.needsUpdateFromModel = false
+                
+                self.model.applyTo(view: self)
+            }
+        }
     }
     
-    override func layoutSubviews() {
-        super.layoutSubviews()
-        contentView.frame = bounds
+    
+    fileprivate struct ViewModel : Equatable {
```

Comment:
We utilize a `ViewModel` instead of simply passing through the `Box`, since the `Box's` wrapped element could contain long lived closures, etc, so want to avoid capturing that context.